packages/capacitor/src/hooks/useAppState.ts

```diff
@@ -0,0 +1,38 @@
+import { App } from '@capacitor/app'
+import { useEffect, useState } from 'react'
+
+import type {
+  RestoredListenerEvent,
+  URLOpenListenerEvent,
+} from '@capacitor/app'
+
+export function useAppState() {
+  const [active, setActive] = useState<boolean>(true)
+  const [urlTrigger, setUrlTrigger] = useState<URLOpenListenerEvent>()
+  const [restoredData, setRestoredData] = useState<RestoredListenerEvent>()
+
+  useEffect(() => {
+    App.addListener('appStateChange', ({ isActive }) => {
+      setActive(isActive)
+    })
+
+    App.addListener('appUrlOpen', (data) => {
+      setUrlTrigger(data)
+    })
+
+    App.addListener('appRestoredResult', (data) => {
+      setRestoredData(data)
+    })
+
+    return () => {
+      // There is no option to remove a single listener. This will remove all listeners!
```

Comment:
It is possible to remove the listener: https://github.com/capacitor-community/react-hooks/blob/bfdf2b7b78651fd1f7c1ce53d4986b7b3148b0ee/packages/app/src/useApp.ts#L28

Reply:
👍